### PR TITLE
simplify some proofs in Slicing.lean

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
*Issue #, if available:*

*Description of changes:*
simplify some proofs in Slicing.lean

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by khieta — APPROVED** [review]

File: cedar-lean/Cedar/Thm/Slicing.lean [hunk 1]
```diff
@@ -166,21 +163,20 @@ theorem scope_bound_is_sound (policy : Policy) :
...
 
 /--
 Scope-based bound analysis is sound.
 -/
 theorem scope_analysis_is_sound :
   IsSoundBoundAnalysis scopeAnalysis
 := by
-  unfold IsSoundBoundAnalysis
-  apply scope_bound_is_sound
+  simp [IsSoundBoundAnalysis, scope_bound_is_sound]
```
Comment (cdisselkoen) [inline cedar-lean/Cedar/Thm/Slicing.lean:179]: is the new version easier to read?  It seems to me the old version is more explicit and perhaps easier to understand what it's doing / why it works

Comment (andrewmwells-amazon) [reply cedar-lean/Cedar/Thm/Slicing.lean:179]: Agree this case is arguable. My preferred proof method is spamming the `simp` tactic and under that metric it's better, but I'm fine with removing this change.

Comment (khieta) [reply cedar-lean/Cedar/Thm/Slicing.lean:179]: +1 for spamming `simp`

Comment (andrewmwells-amazon) [reply cedar-lean/Cedar/Thm/Slicing.lean:179]: I guess let's see what @emina says.

Comment (emina) [reply cedar-lean/Cedar/Thm/Slicing.lean:179]: I also find the more verbose proof perhaps slightly easier to understand, step through, and fix.  But short proofs look nicer :). How about we adopt the shorter proofs, and if we find they are harder to maintain, we go back to the more verbose ones?

**Review by emina — APPROVED** [review]

**Pull request merged**